resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/rest/kubernetes/KubernetesSparkRestServer.scala

```diff
@@ -235,35 +263,39 @@ private[spark] class KubernetesSparkRestServer(
       }
     }
 
-    private def writeUploadedJars(files: Option[TarGzippedData], rootTempDir: File):
+    private def writeUploadedJars(jars: TarGzippedData, rootTempDir: File):
         Seq[String] = {
       val resolvedDirectory = new File(rootTempDir, "jars")
       if (!resolvedDirectory.mkdir()) {
         throw new IllegalStateException(s"Failed to create jars dir at " +
           resolvedDirectory.getAbsolutePath)
       }
-      writeBase64ContentsToFiles(files, resolvedDirectory)
+      CompressionUtils.unpackAndWriteCompressedFiles(jars, resolvedDirectory)
     }
 
-    private def writeUploadedFiles(files: Option[TarGzippedData]): Seq[String] = {
+    private def writeUploadedFiles(files: TarGzippedData): Seq[String] = {
       val workingDir = Paths.get("").toFile.getAbsoluteFile
-      writeBase64ContentsToFiles(files, workingDir)
+      CompressionUtils.unpackAndWriteCompressedFiles(files, workingDir)
     }
 
-    def resolvedAppResource(appResource: AppResource, tempDir: File): String = {
-      val appResourcePath = appResource match {
+    // Retrieve the path on the driver container where the main app resource is, and what value it
+    // ought to have in the spark.jars property. The two may be different because for non-local
+    // dependencies, we have to fetch the resource (if it is not "local") but still want to use
+    // the full URI in spark.jars.
+    private def resolvedAppResource(appResource: AppResource, tempDir: File): (String, String) = {
```

Comment:
I think this description is big enough for javadoc and especially to clarify what the `(String, String)` return value means.  First is local absolute path and second is the value to use in `spark.jars` on driver launch?